File: clinic/src/tumor-types/dto/create-tumor-type.dto.ts
/* eslint-disable prettier/prettier */
import { ApiProperty } from '@nestjs/swagger';
import {IsNotEmpty, IsOptional, IsString, MaxLength, MinLength,} from 'class-validator';

export class CreateTumorTypeDto {
  @ApiProperty({
    description: 'Nombre del tipo de tumor',
    example: 'Cáncer de mama',
    maxLength: 150,
    minLength: 3,
  })
  @IsString()
  @IsNotEmpty({ message: 'El nombre del tipo de tumor es obligatorio' })
  @MinLength(3, { message: 'El nombre debe tener al menos 3 caracteres' })
  @MaxLength(150, { message: 'El nombre no puede exceder 150 caracteres' })
  name: string;

  @ApiProperty({
    description: 'Sistema o área del cuerpo afectada',
    example: 'Glándulas mamarias',
    maxLength: 150,
  })
  @IsString()
  @IsNotEmpty({ message: 'El sistema afectado es obligatorio' })
  @MaxLength(150, {
    message: 'El sistema afectado no puede exceder 150 caracteres',
  })
  systemAffected: string;

  @ApiProperty({
    description: 'Descripción adicional sobre el tipo de tumor',
    example: 'Tumor maligno que se desarrolla en el tejido mamario',
    required: false,
  })
  @IsOptional()
  @IsString()
  @MaxLength(1000, {
    message: 'La descripción no puede exceder 1000 caracteres',
  })
  description?: string;
}

File: auth-gateway/src/main/java/com/genosentinel/auth_gateway/dto/nestjs/CreateTumorTypeDto.java
package com.genosentinel.auth_gateway.dto.nestjs;

import io.swagger.v3.oas.annotations.media.Schema;
import jakarta.validation.constraints.NotBlank;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Schema(description = "Datos para crear un tipo de tumor")
public class CreateTumorTypeDto {

    @NotBlank(message = "El nombre es obligatorio")
    @Schema(description = "Nombre del tipo de tumor", example = "Melanoma")
    private String name;

    @Schema(description = "El tipo de tumor", example = "Glándulas mamarias")
    private String systemAffected;

    @Schema(description = "Descripción del tipo de tumor", example = "Tipo de cáncer de piel")
    private String description;
}